src/Backups/BackupIO_S3.cpp

```diff
@@ -60,6 +64,7 @@ namespace
         client_configuration.connectTimeoutMs = 10 * 1000;
         /// Requests in backups can be extremely long, set to one hour
         client_configuration.requestTimeoutMs = 60 * 60 * 1000;
```

Comment:
It is bad decision. Your request may just stack for nor reason for 1h.
See https://github.com/ClickHouse/clickhouse-private/issues/3311

Reply:
As far as I remember that timeout was written because a backup can be long and it's sad to throw away all the work after a small failure near the end.

Reply:
requestTimeoutMs -- it is not a request timeout actually. We do not have timeout for the whole request.
That timeout determine send/receive one call timeout. That operation are not suppose to be long. Usually it is 10-200ms. In exceptional cases (complete multi part) I saw it as 10-20s.

>  it's sad to throw away all the work after a small failure near the end. 

I tottaly agree with that.
My suggestion is to set requestTimeoutMs as 30s. Use aggressive timeouts from this https://github.com/ClickHouse/ClickHouse/pull/56314. 
Hovewer, the most important part is significantly increasing retries count.

Reply:
Ok, I agree.